Source/JavaScriptCore/CMakeLists.txt

```
@@ -351,51 +351,85 @@ set(BUILTINS_GENERATOR_SCRIPTS
-    ${JAVASCRIPTCORE_DIR}/builtins/SetPrototype.js
-    ${JAVASCRIPTCORE_DIR}/builtins/ShadowRealmPrototype.js
-    ${JAVASCRIPTCORE_DIR}/builtins/StringConstructor.js
-    ${JAVASCRIPTCORE_DIR}/builtins/StringIteratorPrototype.js
-    ${JAVASCRIPTCORE_DIR}/builtins/StringPrototype.js
-    ${JAVASCRIPTCORE_DIR}/builtins/TypedArrayConstructor.js
-    ${JAVASCRIPTCORE_DIR}/builtins/TypedArrayPrototype.js
-    ${JAVASCRIPTCORE_DIR}/builtins/WebAssembly.js
-    ${JAVASCRIPTCORE_DIR}/inspector/InjectedScriptSource.js
+set(JavaScriptCore_BUILTINS_SOURCE_NAMES
+    builtins/ArrayConstructor.js
+    builtins/ArrayIteratorPrototype.js
+    builtins/ArrayPrototype.js
+    builtins/AsyncFromSyncIteratorPrototype.js
+    builtins/AsyncFunctionPrototype.js
+    builtins/AsyncGeneratorPrototype.js
+    builtins/AsyncIteratorPrototype.js
+    builtins/DatePrototype.js
+    builtins/FunctionPrototype.js
+    builtins/GeneratorPrototype.js
+    builtins/GlobalObject.js
+    builtins/GlobalOperations.js
+    builtins/InternalPromiseConstructor.js
+    builtins/IteratorHelpers.js
+    builtins/IteratorPrototype.js
+    builtins/MapIteratorPrototype.js
+    builtins/MapPrototype.js
+    builtins/ModuleLoader.js
+    builtins/NumberConstructor.js
+    builtins/ObjectConstructor.js
+    builtins/PromiseConstructor.js
+    builtins/PromiseOperations.js
+    builtins/PromisePrototype.js
+    builtins/ProxyHelpers.js
+    builtins/ReflectObject.js
+    builtins/RegExpPrototype.js
+    builtins/RegExpStringIteratorPrototype.js
+    builtins/SetIteratorPrototype.js
+    builtins/SetPrototype.js
+    builtins/ShadowRealmPrototype.js
+    builtins/StringConstructor.js
+    builtins/StringIteratorPrototype.js
+    builtins/StringPrototype.js
+    builtins/TypedArrayConstructor.js
+    builtins/TypedArrayPrototype.js
+    builtins/WebAssembly.js
+
+    inspector/InjectedScriptSource.js
 )
 
+set(JavaScriptCore_BUILTINS_SOURCES)
+set(JavaScriptCore_PREPROCESSED_BUILTINS_SOURCES)
+file(MAKE_DIRECTORY ${JavaScriptCore_DERIVED_SOURCES_DIR}/preprocessed)
+foreach (_file ${JavaScriptCore_BUILTINS_SOURCE_NAMES})
+    get_filename_component(_directory ${_file} DIRECTORY)
+    file(MAKE_DIRECTORY ${JavaScriptCore_DERIVED_SOURCES_DIR}/preprocessed/${_directory})
+    if (MSVC)
+        add_custom_command(
+            OUTPUT ${JavaScriptCore_DERIVED_SOURCES_DIR}/preprocessed/${_file}
+            MAIN_DEPENDENCY ${JAVASCRIPTCORE_DIR}/${_file}
+            DEPENDS ${WTF_INTERFACE_DEPENDENCIES} ${JAVASCRIPTCORE_DIR}/builtins/JSCBuiltinConfig.h
+            COMMAND ${CMAKE_C_COMPILER} /P /EP /FI${JAVASCRIPTCORE_DIR}/builtins/JSCBuiltinConfig.h /I${WTF_FRAMEWORK_HEADERS_DIR} /I${bmalloc_FRAMEWORK_HEADERS_DIR} ${JAVASCRIPTCORE_DIR}/${_file} /Fi${JavaScriptCore_DERIVED_SOURCES_DIR}/preprocessed/${_file}
+            VERBATIM)
+    else ()
+        add_custom_command(
+            OUTPUT ${JavaScriptCore_DERIVED_SOURCES_DIR}/preprocessed/${_file}
```

nit: `.i` is the typical file extension for preprocessed C source, would it be difficult to use it here too? We could do:
```suggestion
            OUTPUT ${JavaScriptCore_DERIVED_SOURCES_DIR}/preprocessed/${_file}.i
```